cmd/serverless-init/main.go

```diff
@@ -8,108 +8,126 @@
-	initcontainer.Run(cloudService, logConfig, metricAgent, traceAgent, logsAgent, cliParams.args)
-}
+// removing these unused dependencies will cause silent crash due to fx framework
+func run(_ secrets.Component, _ autodiscovery.Component, _ healthprobeDef.Component) {
+	cloudService, logConfig, traceAgent, metricAgent, logsAgent := setup(modeConf)
 
-func setup() (cloudservice.CloudService, *log.Config, *trace.ServerlessTraceAgent, *metrics.ServerlessMetricAgent, logsAgent.ServerlessLogsAgent) {
-	if err := config.SetupLogger(
-		loggerName,
-		"error", // will be re-set later with the value from the env var
-		"",      // logFile -> by setting this to an empty string, we don't write the logs to any file
-		"",      // syslog URI
-		false,   // syslog_rfc
-		true,    // log_to_console
-		false,   // log_format_json
-	); err != nil {
-		logger.Errorf("Unable to setup logger: %s", err)
-	}
+	modeConf.Runner(logConfig)
 
-	if logLevel := os.Getenv(logLevelEnvVar); len(logLevel) > 0 {
-		if err := config.ChangeLogLevel(logLevel); err != nil {
-			logger.Errorf("Unable to change the log level: %s", err)
-		}
-	}
+	metric.AddShutdownMetric(cloudService.GetPrefix(), metricAgent.GetExtraTags(), time.Now(), metricAgent.Demux)
+	lastFlush(logConfig.FlushTimeout, metricAgent, traceAgent, logsAgent)
+}
 
+func setup(mode.Conf) (cloudservice.CloudService, *serverlessInitLog.Config, *trace.ServerlessTraceAgent, *metrics.ServerlessMetricAgent, logsAgent.ServerlessLogsAgent) {
 	tracelog.SetLogger(corelogger{})
 
 	// load proxy settings
-	config.LoadProxyFromEnv(config.Datadog())
+	pkgconfig.LoadProxyFromEnv(pkgconfig.Datadog())
 
 	cloudService := cloudservice.GetCloudServiceType()
 
+	log.Debugf("Detected cloud service: %s", cloudService.GetOrigin())
+
 	// Ignore errors for now. Once we go GA, check for errors
 	// and exit right away.
 	_ = cloudService.Init()
 
-	tags := tags.MergeWithOverwrite(tags.ArrayToMap(configUtils.GetConfiguredTags(config.Datadog(), false)), cloudService.GetTags())
+	tags := serverlessInitTag.GetBaseTagsMapWithMetadata(
+		serverlessTag.MergeWithOverwrite(
+			serverlessTag.ArrayToMap(configUtils.GetConfiguredTags(pkgconfig.Datadog(), false)),
+			cloudService.GetTags()),
+		modeConf.TagVersionMode)
+
 	origin := cloudService.GetOrigin()
 	prefix := cloudService.GetPrefix()
 
-	logConfig := log.CreateConfig(origin)
-	logsAgent := log.SetupLog(logConfig, tags)
-
-	// Disable remote configuration for now as it just spams the debug logs
-	// and provides no value.
-	os.Setenv("DD_REMOTE_CONFIGURATION_ENABLED", "false")
+	agentLogConfig := serverlessInitLog.CreateConfig(origin)
 
 	// The datadog-agent requires Load to be called or it could
 	// panic down the line.
-	_, err := config.LoadWithoutSecret()
+	_, err := pkgconfig.LoadWithoutSecret()
 	if err != nil {
-		logger.Debugf("Error loading config: %v\n", err)
+		log.Debugf("Error loading config: %v\n", err)
 	}
```

Comment:
Since you are using Fx to load certain components and one of them being the config component, I believe you have access to the config object through the config component, so we could remove this line and use the config component down the stack. 

This probably requires a bit of refactoring in the different elements of the serverless init, which would require passing the config component to the different functions. Such as `setupTraceAgent`, `setupMetricAgent ` and any other function that uses the config.

Let me know if it clear and if not I'm more than happy to pair to on it.